packages/backend.ai-ui/src/components/baiClient/FileExplorer/BAIFileExplorer.tsx

```diff
@@ -0,0 +1,334 @@
+import { BAIFileExplorerFragment$key } from '../../../__generated__/BAIFileExplorerFragment.graphql';
+import {
+  convertToDecimalUnit,
+  filterEmptyItem,
+  localeCompare,
+} from '../../../helper';
+import { BAITrashBinIcon } from '../../../icons';
+import BAIFlex from '../../BAIFlex';
+import { BAITable } from '../../Table';
+import { VFolderFile } from '../../provider/BAIClientProvider/types';
+import ActionItems from './ActionItems';
+import ControlItems from './ControlItems';
+import DeleteSelectedItemsModal from './DeleteSelectedItemsModal';
+import { useSearchVFolderFiles } from './hooks';
+import { FileOutlined, FolderOutlined, HomeOutlined } from '@ant-design/icons';
+import { Breadcrumb, Button, TableColumnsType, theme, Typography } from 'antd';
+import { createStyles } from 'antd-style';
+import { ItemType } from 'antd/es/breadcrumb/Breadcrumb';
+import dayjs from 'dayjs';
+import _ from 'lodash';
+import { DownloadIcon } from 'lucide-react';
+import { createContext, Suspense, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { graphql, useFragment } from 'react-relay';
+
+const useStyles = createStyles(({ css, token }) => ({
+  hover: css`
+    text-decoration: none;
+    /* color: ${token.colorLink}; */
+
+    &:hover {
+      /* color: ${token.colorLinkHover}; */
+      text-decoration: underline;
+    }
+  `,
+}));
+
+export const FolderInfoContext = createContext<{
+  targetVFolderId: string;
+  currentPath: string;
+}>({
+  targetVFolderId: '',
+  currentPath: '.',
+});
+
+export interface BAIFileExplorerProps {
+  vfolderNodeFrgmt?: BAIFileExplorerFragment$key | null;
+  targetVFolderId: string;
+}
+
+const BAIFileExplorer: React.FC<BAIFileExplorerProps> = ({
+  vfolderNodeFrgmt,
+  targetVFolderId,
+}) => {
+  const { t } = useTranslation();
+  const { token } = theme.useToken();
+  const { styles } = useStyles();
+  const [selectedItems, setSelectedItems] = useState<Array<VFolderFile>>([]);
+  const [selectedSingleItem, setSelectedSingleItem] =
+    useState<VFolderFile | null>(null);
+  const [breadcrumbItems, setBreadcrumbItems] = useState<Array<ItemType>>([
+    { title: <HomeOutlined />, onClick: () => navigateToPath('.') },
+  ]);
+  const [pathFoldersCache, setPathFoldersCache] = useState<
+    Record<string, Array<VFolderFile>>
+  >({});
+  const {
+    files,
+    isFetching,
+    currentPath,
+    navigateDown,
+    navigateToPath,
+    refetch,
+  } = useSearchVFolderFiles(targetVFolderId);
+  const [cachedFetchedFiles, setCachedFetchedFiles] = useState<
+    Array<VFolderFile>
+  >([]);
+
+  useEffect(() => {
+    if (!_.isNil(files?.items)) {
+      setCachedFetchedFiles(files.items);
+    }
+  }, [files]);
+
+  const FileExplorerFragment = useFragment(
```

Comment:
The variable should start with a lowercase letter, and since the return value of `useFragment` is not a fragment or FileExplorer but just a VirtualFolder, please rename the variable to `virtualFolder`, `vFolder`, or `vFolderNode`.